tendrl/commons/objects/node/atoms/check_node_up/__init__.py

```diff
@@ -1,40 +1,81 @@
+import etcd
+import json
+
+from tendrl.commons.flows.exceptions import FlowExecutionFailedError
 from tendrl.commons.objects import BaseAtom
 from tendrl.commons.utils.cmd_utils import Command
+from tendrl.commons.utils import etcd_utils
 from tendrl.commons.utils import log_utils as logger
 
 
 class CheckNodeUp(BaseAtom):
+    def __init__(self, *args, **kwargs):
+        super(CheckNodeUp, self).__init__(*args, **kwargs)
 
     def run(self):
-        fqdn = self.parameters.get("fqdn")
-        logger.log(
-            "info",
-            NS.publisher_id,
-            {"message": "Checking if node %s is up" %
-             self.parameters.get("fqdn")},
-            job_id=self.parameters['job_id'],
-            flow_id=self.parameters['flow_id']
-        )
-        out, err, rc = Command("ping -c 1 %s" % fqdn).run()
-        # and then check the response...
-        if not err and rc == 0:
-            logger.log(
-                "info",
-                NS.publisher_id,
-                {"message": "Ping to %s succeeded." %
-                 self.parameters.get("fqdn")},
-                job_id=self.parameters['job_id'],
-                flow_id=self.parameters['flow_id']
-            )
-            return True
-        else:
-            logger.log(
-                "info",
-                NS.publisher_id,
-                {"message": "Failed to ping %s. Error %s" % (
-                    self.parameters.get("fqdn"),
-                    err)},
-                job_id=self.parameters['job_id'],
-                flow_id=self.parameters['flow_id']
+        try:
+            flag = True
+            # check job is parent or child
+            job = NS.tendrl.objects.Job(
+                job_id=self.parameters['job_id']
+            ).load()
+            if "parent" not in job.payload:
+                # Getting node_ids from parameters
+                node_ids = self.parameters.get(
```

Comment:
you are not declaring "Node[]" as mandatory or optional inputs, why are you trying to fetch it here?

Reply:
i will remove it